### PR TITLE
Awesome books - Refactoring for Class Structure

### PR DESCRIPTION
**Pull request by yigitm**
We add the below features:

- Changing content to Class structure,
- Insert 'add', 'show' and 'remove' static methods,
- Insert validation for 'empty input check',

### CODE REVIEW

File: js/mainclass.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
+let books = [];
+const form = document.getElementById('form-books');
+const divCont = document.getElementById('display-books');
+
+class Library {
+  constructor(title, author) {
+    this.title = title;
+    this.author = author;
+    this.id = parseInt(new Date().getTime(), 10);
+  }
```
Comment (D4nielJ) [inline js/mainclass.js:10]: - [ ] Let's change the constructor of the Library to be `this.books = []`.

[OPTIONAL] You can create a new class called Book with the constructor you're using in the Library class.

```
class Book {
  constructor(title, author) {
    this.title = title;
    this.author = author;
    this.id = parseInt(new Date().getTime(), 10);
  }
```

File: js/mainclass.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  author.value = '';
+  title.focus();
+});
+
+divCont.addEventListener('click', (e) => {
+  Library.removeBook(e.target.id);
+});
+
+Library.getItems();
+Library.showInitial();
```
Comment (D4nielJ) [inline js/mainclass.js:85]: - [ ] For the library class to be able to use its books array, you will need to create an instance of the class like this: `const library = new Library()`

Once you have correctly instantiated the class you will be able to access its methods `library.getItems();` or `library.showInitial();` but you won't need them to be `static` methods.

File: js/mainclass.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+class Library {
+  constructor(title, author) {
+    this.title = title;
+    this.author = author;
+    this.id = parseInt(new Date().getTime(), 10);
+  }
+
+  // set Local
+  static setLocal(books) {
```
Comment (D4nielJ) [inline js/mainclass.js:13]: - [ ] You will need in any place where you call the variable `books` for `this.books`

File: js/mainclass.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    localStorage.setItem('LibraryBooks', JSON.stringify(books));
+  }
+
+  // Get items
+  static getItems() {
+    if (!localStorage.getItem('LibraryBooks')) {
+      books = [];
+      Library.setLocal();
```
Comment (D4nielJ) [inline js/mainclass.js:21]: - [ ] I think that to call on methods from the class you could use `this.setLocal();`

**Review by D4nielJ — CHANGES_REQUESTED** [review]
## Hello @yigitm and @iwillteixeira, 

You're doing great! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

## Happy coding and stay awesome! 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can always find me on slack!

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: js/mainclass.js (hunk 4 above)
Comment (yigitm) [reply js/mainclass.js]: Thank you for the suggestion. We changed it.

File: js/mainclass.js (hunk 3 above)
Comment (yigitm) [reply js/mainclass.js]: Thank you for the suggestion. We changed it.

File: js/mainclass.js (hunk 2 above)
Comment (yigitm) [reply js/mainclass.js]: Thank you for the suggestion. We changed it.

File: js/mainclass.js (hunk 1 above)
Comment (yigitm) [reply js/mainclass.js]: Thank you for the suggestion. We changed it.

Comment (henatan99) [reply js/mainclass.js]: This issue has been addressed well. Good job! :clap: :+1:

File: js/mainclass.js (hunk 2 above)
Comment (henatan99) [reply js/mainclass.js]: This issue has been addressed well. Great job! :clap: :clap: :clap:

File: js/mainclass.js (hunk 4 above)
Comment (henatan99) [reply js/mainclass.js]: This issue has been addressed well. Nice job! :pray: :pray:

**Review by henatan99 — APPROVED** [review]
Hi @yigitm , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**